modules/charts/src/components/Chart.tsx

```diff
@@ -23,10 +23,23 @@ type ChartProps = {
 	DisplayComponent?: React.ReactElement<ArrangerChartProps>;
 };
 
-// TODO: numeric or agg, very hacky property check, no need for this once server config is expanded, then we can check the typename
-const resolveData = ({ data }) => {
-	if (data.buckets) return data.buckets;
-	if (data.range) return data.range.buckets;
+// TODO: numeric or agg, very hacky property check,
+//  no need for this once server config is expanded, then we can check the typename
+const resolveData = ({ data, onDataLoad }) => {
+	let chartData = {};
+
+	if (data.buckets) {
+		chartData = data.buckets;
+	} else if (data.range) {
+		chartData = data.range.buckets;
+	}
+
+	// consumer
+	if (onDataLoad) {
+		chartData = onDataLoad(chartData);
+	}
+
+	return chartData;
 };
```

Comment:
Unclear what the data type returned here is. I'm sure this is inferred correctly but for a reader its very unclear without function signature.